unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/Tests/GltfContainerHandlerShould.cs

```diff
@@ -0,0 +1,186 @@
+        private DataStore_ECS7 dataStoreEcs7;
+
+        [SetUp]
+        public void SetUp()
+        {
+            ServiceLocator serviceLocator = ServiceLocatorTestFactory.CreateMocked();
+            serviceLocator.Register<IWebRequestController>(WebRequestController.Create);
+            Environment.Setup(serviceLocator);
+
+            ECSComponentsFactory componentFactory = new ECSComponentsFactory();
+            ECSComponentsManager componentsManager = new ECSComponentsManager(componentFactory.componentBuilders);
+            InternalECSComponents internalEcsComponents = new InternalECSComponents(componentsManager, componentFactory);
+
+            testUtils = new ECS7TestUtilsScenesAndEntities(componentsManager);
+            scene = testUtils.CreateScene("temptation");
+            entity = scene.CreateEntity(23423);
+
+            var keepEntityAliveComponent = new InternalECSComponent<object>(
+                0, componentsManager, componentFactory, null);
+            keepEntityAliveComponent.PutFor(scene, entity, entity);
+
+            scene.contentProvider.baseUrl = $"{TestAssetsUtils.GetPath()}/GLB/";
+            scene.contentProvider.fileToHash.Add("palmtree", "PalmTree_01.glb");
+            scene.contentProvider.fileToHash.Add("sharknado", "Shark/shark_anim.gltf");
+
+            renderersComponent = internalEcsComponents.renderersComponent;
+            pointerColliderComponent = Substitute.For<IInternalECSComponent<InternalColliders>>();
+            physicColliderComponent = Substitute.For<IInternalECSComponent<InternalColliders>>();
+            dataStoreEcs7 = new DataStore_ECS7();
+
+            handler = new GltfContainerHandler(pointerColliderComponent, physicColliderComponent, renderersComponent, dataStoreEcs7);
+            handler.OnComponentCreated(scene, entity);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            handler.OnComponentRemoved(scene, entity);
+            testUtils.Dispose();
+            AssetPromiseKeeper_GLTF.i.Cleanup();
+            PoolManager.i.Dispose();
+        }
+
+        [UnityTest]
+        public IEnumerator UpdateCorrectly()
+        {
+            PBGltfContainer model = new PBGltfContainer()
+            {
+                Src = "palmtree"
+            };
+
+            handler.OnComponentModelUpdated(scene, entity, model);
+            yield return new WaitUntil(() => handler.gltfLoader.isFinished);
+
+            Assert.AreEqual("PalmTree_01", handler.gameObject.transform.GetChild(0).GetChild(0).name);
+
+            model = new PBGltfContainer()
+            {
+                Src = "sharknado"
+            };
+
+            handler.OnComponentModelUpdated(scene, entity, model);
+            yield return new WaitUntil(() => handler.gltfLoader.isFinished);
+
+            Assert.AreEqual("shark_skeleton", handler.gameObject.transform.GetChild(0).GetChild(0).GetChild(0).name);
```

Comment:
I would add a comment here for future devs to understand why we are navigating through child 3 times, what if the hierarchy changes? ( It's going to change with GLTFast importer )

Reply:
If GLTF hierarchy changes then the test must be changed accordingly